docker-compose.yml

```diff
@@ -31,10 +31,9 @@
 # See https://docs.mathesar.org/configuration/env-variables/ for more info.
 #
 x-config: &config
-  # (REQUIRED) Replace '?' with '-' followed by a 50 character random string.
-  # You can generate one at https://djecrety.ir/ or by running:
-  #   echo $(cat /dev/urandom | LC_CTYPE=C tr -dc 'a-zA-Z0-9' | head -c 50)
-  SECRET_KEY: ${SECRET_KEY:?}
+  # (Not recommended) To use a custom secret key, use an .env file, or add a
```

Comment:
As an existing user (Not recommended) would make me think that there is something wrong with me having my own secret key configured and/or it is now deprecated.

I suggest moving `SECRET_KEY` from `x-config` to the `environment` section for `mathesar_service` container so that we can ensure backwards compatibility. New users don't need to worry about adding one at all.

We could move the documentation for secret key within the `environment` section as well. 

Also, the "HOW TO USE THIS FILE" section mentions the need to add a secret key before running docker compose which needs to removed.

Reply:
Okay, reorganized.